runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollEmoteOverlay.java

```diff
@@ -0,0 +1,156 @@
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.cluescrolls;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.FontMetrics;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.plugins.cluescrolls.clues.EmoteClue;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.TextComponent;
+
+public class ClueScrollEmoteOverlay extends Overlay
+{
+	private static final Color HIGHLIGHT_BACKGROUND = new Color(0, 255, 0, 100);
+
+	private final ClueScrollPlugin plugin;
+	private final Client client;
+	private final TextComponent textComponent = new TextComponent();
+
+	@Inject
+	public ClueScrollEmoteOverlay(ClueScrollPlugin plugin, Client client)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.plugin = plugin;
+		this.client = client;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics, Point parent)
+	{
+		Object clue = plugin.getClue();
+
+		if (clue == null)
+		{
+			return null;
+		}
+
+		if (clue instanceof EmoteClue)
+		{
+			EmoteClue emoteClue = (EmoteClue) clue;
+
+			if (!emoteClue.getFirstEmote().hasSprite())
+			{
+				return null;
+			}
+
+			Widget emoteContainer = client.getWidget(WidgetInfo.EMOTE_CONTAINER);
+
+			if (emoteContainer == null)
+			{
+				return null;
+			}
+
+			if (emoteContainer.isHidden())
+			{
+				return null;
+			}
+
+			Widget emoteWindow = client.getWidget(WidgetInfo.EMOTE_WINDOW);
+
+			if (emoteWindow == null)
+			{
+				return null;
+			}
+
+			for (Widget emoteWidget : emoteContainer.getDynamicChildren()) 
```

Comment:
This can be slightly improved to use WidgetHiddenChanged event, it fires for every emote when you open the emote window.